### PR TITLE
Documentation sprint work: first attempt at user docs on objects, media, and context.

### PR DESCRIPTION
**Pull request by mjordan**
Github issue: https://github.com/Islandora-CLAW/CLAW/issues/1040

# What does this Pull Request do?

This PR is part of the March 2019 documentation sprint, and includes work on the user docs for objects, media, and context as outlined at https://docs.google.com/spreadsheets/d/1wI_K5Y0XJSIBXoJ84VxObcy8TT07TuypIE_P3C0rnzU/edit#gid=0. 

Updated or new pages can be viewed at:

* [objects](https://github.com/mjordan/CLAW/blob/mjordandocsprint/docs/user-documentation/objects.md)
* [media](https://github.com/mjordan/CLAW/blob/mjordandocsprint/docs/user-documentation/datastreams.md)
* [context](https://github.com/mjordan/CLAW/blob/mjordandocsprint/docs/user-documentation/context.md)

# Interested parties
@cdeaneGit @cderven

### CODE REVIEW

**Comment by cdeaneGit** [comment]
Partial review focused on:
 1.  Media in Islandora 8
 2. What is an Islandora 8 object?
Everything well written and very helpful.

File: docs/user-documentation/datastreams.md [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+
+![Media tab](../assets/media_tab.png)
+
+For this image, we can see that a "Service File" and a "Thumbnail Image" derivatives have been generated and added to the object's list of media:
+
+![Media tab](../assets/islandora_8_derivatives_sample.png)
+
+### Media revisions
+
+To be complted on resolutino of https://github.com/Islandora-CLAW/CLAW/issues/1035.
```
Comment (dannylamb) [inline docs/user-documentation/datastreams.md]: s/resolutino/resolution

Comment (mjordan) [reply docs/user-documentation/datastreams.md]: 😄

Comment (mjordan) [reply docs/user-documentation/datastreams.md]: Also, s/complted /completed fixed.

File: docs/user-documentation/objects.md [hunk 2]
```diff
@@ -1,26 +1,65 @@
...
-    * It’s a node, with maybe one or more medias and files attached (see “Datastreams are Media” section)
-    * The node has metadata in its fields (See "Metadata are Fields" section.)
-    * It has a URI, and a JSON-LD view in Drupal
-    * It goes into Fedora. (See todo make "In Fedora" section)
-    * By Drupal default, all published nodes are public, so see todo make Access Control section.
-    * As a Drupal node, it’s available to Views and gets indexed in Solr.
-    * We suggest using a taxonomy term on the node to control elements of its display or editing, e.g. in Islandora Demo there is a taxonomy term that sets the type: Audio, Binary, Collection, Image, and Video
-    * How do I make an Islandora object? (Make a node, then add a Media)
+```
+userid: 1
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: s/userid/uid

File: docs/user-documentation/context.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+Let's look at the example of how Context can determine how an object is diplayed. Drupal 8 has the idea of "view modes", which allow site builders to choose what happens when an object is viewed by the user (it has nothing to do with Drupal Views). In the node edit form for Islandora objects, there is a checkbox that, if checked, tells Drupal to render the image using the OpenSeadragon viewer:
+
+![Display Hints field in node edit form](../assets/context_display_hints.png)
+
+ This  functionality is accomplished via the "Open Seadragon" Context, which, as its configuration suggests, checks as its condition whether the node as the "Open Seadragon" tag and if so, reacts by using the view mode "Open Seadragon":
+
+![Open Seadragon Context configuration](../assets/context_openseadragon_configuration.png)
+
+### Context and Actions
```
Comment (Natkeeran) [inline docs/user-documentation/context.md]: May want to discuss how custom actions can be defined.  Most of the actions used by Islandora is defined using Plugins.

File: docs/user-documentation/objects.md [hunk 4]
```diff
@@ -1,26 +1,65 @@
...
+
+Islandora 7.x basically inherits its object model from Fedora 3.x. In 7.x, Fedora stores all properties and content associated with an object - not only its owner, dc.title, status, PID, and status, but also any content files such as OBJ, DC, MODS, and RELS-EXT. In Islandora 7.x, Fedora is the authoritative, primary source for all aspects of an object. Fedora 3.x is not an optional component of an Islandora 7.x repository, it is the primary datastore.
+
+In Islandora 8, using Fedora is optional. That's right, optional. Drupal, and not Fedora, is the primary source of all aspects of an Islandora 8 object, and, with some variations, Drupal, not Fedora, is the primary datastore in an Islandora repository. If Fedora is present in an Islandora 8 repository, content in it is a tightly syncronized copy of object properties and files managed by Drupal.
+
+Even though Fedora is optional in Islandora 8, most repositories will use it since it provides its own set of services that are worth taking advantage of, such as:
+
+* flexible, and configurable, disk storage architecture
+* fixity digest generation
+* Momento versioning
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: s/Momento/Memento

File: docs/user-documentation/objects.md [hunk 5]
```diff
@@ -1,26 +1,65 @@
...
+* integration with RDF/Linked Data triplestores
+* integration with microservices
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: Integration with Microservices via API-X

File: docs/user-documentation/objects.md [hunk 6]
```diff
@@ -1,26 +1,65 @@
...
+* access control mechanisms
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: WebAC Policies for access control

Comment (mjordan) [reply docs/user-documentation/objects.md]: Nice catch @dannylamb

File: docs/user-documentation/objects.md [hunk 7]
```diff
@@ -1,26 +1,65 @@
...
+content model | type (and also tags from the Islandora Models vocabulary)
+status | status
+
+In addition to these basic node properties, Islandora 8 objects (like all Drupal nodes) can have fields, which is where most of what we would think of as metadata is stored. The section on [metadata](metadata.md) describes how fields work.
+
+Islandora's 7.x's content models do not exist in Islandora 8. The primary way that Islandora 8 identifies what we think of as a conetent model in Islandora 7.x is using taxonomy terms from the "Islandora Models" vocabulary. To indicate that an Islandora 8 object is an image, video, etc, the user selects a model from a select list:
+
+![Media tab](../assets/object_model_tags.png)
+
+The combination of Drupal content type (e.g., islandora_object) and a value from the "Islandora Models" vocabulary (e.g. "Image") are the equivalent of Islandora 7.x's content models. In Islandora 7.x, an object's content model is immutable, but in Islandora 8, an object's model can be changed from within its node edit form using a select list.
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: Might be worth making the distinction that the content type is more like the metadata profile, and the models vocab tells you what type of object you have.  E.g. and Collection is a Collection no matter the content_type/metadata standard that gets used, so long as the node is tagged with the Collection term.

Comment (mjordan) [reply docs/user-documentation/objects.md]: Good suggestion, I'll take care of that this evening.

File: docs/user-documentation/objects.md (hunk 2 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: Done.

File: docs/user-documentation/context.md (hunk 3 above)
Comment (mjordan) [reply docs/user-documentation/context.md]: Thanks @Natkeeran, I'll do that this evening.

File: docs/user-documentation/objects.md (hunk 4 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: Done.

File: docs/user-documentation/objects.md (hunk 5 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: Done.

File: docs/user-documentation/objects.md (hunk 6 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: Done.

File: docs/user-documentation/objects.md (hunk 7 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: OK, added a section "Drupal Content Types as Islandora Metadata Profiles".

File: docs/user-documentation/context.md (hunk 3 above)
Comment (mjordan) [reply docs/user-documentation/context.md]: @Natkeeran I've added an example of creating an Action that thanks the user when they create a Media and tag it as a "Preservation Master File." Check it out: https://github.com/mjordan/CLAW/blob/mjordandocsprint/docs/user-documentation/context.md.

**Comment by mjordan** [comment]
@cderven thanks, I just performed a couple of updates, to remove redundant words and to add a couple words for clarity.

File: docs/user-documentation/objects.md [hunk 8]
```diff
@@ -1,26 +1,65 @@
...
+```
 
+These are Drupal node properties, which all Drupal nodes have (although not all Drupal nodes have a type of "islandora_object", some have a type of "page" or "article" or "some_custom_content_type"). These node properties directly correspond to an Islandora 7.x object's properties:
 
-!!! tip "7.x migration note"
-    #### Where are my Solution Packs?
-    If you are familiar with Islandora 7.x, a solution pack is a bundle representing objects of a certain 'type', that are associated with metadata forms, an ingest workflow that allows specific filetypes, derivative settings based on the uploaded object, an optional content viewer, and the look and feel of an object's page including what metadata to display to the user.
+Islandora 7.x object properties | Islandora 8.x node/object properties
+------------ | -------------
+owner | userid
```
Comment (dannylamb) [inline docs/user-documentation/objects.md]: Another s/userid/uid

Sorry i missed this one earlier

**Comment by dannylamb** [comment]
I'm generally :+1: to this.  There's a lot of 'for 7.x' info in here that we'll probably want to move around in a later editorial pass, but let's get this in.  If you can touch up one last `userid` I'll merge this if @cderven and @Natkeeran are satisfied.

File: docs/user-documentation/objects.md (hunk 8 above)
Comment (mjordan) [reply docs/user-documentation/objects.md]: Done.

**Comment by dannylamb** [comment]
Gonna go ahead and move forward with this, and we can address further feedback in subsequent PRs.

**Pull request merged**